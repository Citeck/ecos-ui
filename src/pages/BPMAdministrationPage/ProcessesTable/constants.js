import { BpmnProcessStatusColumn } from '../../../components/BpmnProcessStatusColumn';
import { URL } from '../../../constants';
import { getMLValue } from '../../../helpers/util';

export const PROCESSED_TABLE_COLUMNS = [
  {
    newFormatter: { type: 'link', config: { getUrl: row => `${URL.BPMN_ADMIN_PROCESS}?recordRef=${row.id}` } },
    text: getMLValue({
      ru: 'ID Процесса',
      en: 'Process ID'
    }),
    attribute: 'key',
    dataField: 'key'
  },
  {
    text: getMLValue({
      ru: 'Описание процесса',
      en: 'Process definition'
    }),
    attribute: 'ecosDef',
    dataField: 'ecosDef'
  },
  {
    text: getMLValue({
      ru: 'Количество инцидентов',
      en: 'Incident count'
    }),
    attribute: 'incidentCount',
    dataField: 'incidentCount'
  },
  {
    text: getMLValue({
      ru: 'Количество запущенных экземпляров',
      en: 'Running process count'
    }),
    attribute: 'instancesCount',
    dataField: 'instancesCount'
  },
  {
    classes: 'common-table-column__small',
    newFormatter: { type: 'component', config: { Component: BpmnProcessStatusColumn } },
    text: getMLValue({
      ru: 'Состояние',
      en: 'State'
    }),
    attribute: 'key',
    dataField: 'key'
  }
];
