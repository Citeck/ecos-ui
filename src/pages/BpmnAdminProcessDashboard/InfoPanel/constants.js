export const Labels = {
  INFO_TITLE: 'bpmn-admin.info-title',

  PROCESS_ID: 'bpmn-admin.process-id',
  VERSION: 'bpmn-admin.version',
  DEFINITION_REF: 'bpmn-admin.definition-ref',
  MODIFIED: 'bpmn-admin.modified',
  MODIFIER: 'bpmn-admin.modifier',
  DEPLOYMENT_ID: 'bpmn-admin.deployment-id',
  BY_ALL_VERSIONS: 'bpmn-admin.by-all-versions',
  VERSIONS_COUNT: 'bpmn-admin.versions',
  INSTANCES_COUNT: 'bpmn-admin.instances-count',
  INCIDENT_COUNT: 'bpmn-admin.incident-count',

  NOT_DEFINED: 'bpmn-admin.incident.not-defined'
};
