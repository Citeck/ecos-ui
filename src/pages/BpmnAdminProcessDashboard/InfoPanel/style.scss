@import '../../../styles/constants.scss';

.info-panel {
  &__body {
    padding: 16px;
  }

  &__main {
    display: flex;
    justify-content: space-between;
  }

  & .ecos-btn_blue {
    position: relative;
    padding: 3px 12px;
    overflow: hidden;

    p {
      margin: 0;
      max-width: 80%;
      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;
    }
  }

  & .ecos-btn__content {
    display: flex;
    justify-content: space-between;
  }

  &__main-item {
    display: flex;
    flex-direction: column;
    gap: 8px;

    width: 300px;

    &_full_width {
      width: auto;
      margin-top: 8px;
    }

    &_bottom {
      display: flex;
      align-items: flex-end;
    }
  }

  &__actions {
    display: flex;
    justify-content: space-between;
    width: 300px;
  }

  &__secondary {
    margin: 16px 0;
    color: $color-gray-5;
  }

  &__statistic {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-gap: 8px;

    height: 110px;

    &-item {
      position: relative;
      background: $color-yellow;

      border-radius: 12px;
      padding: 8px;

      & > * {
        margin: 0;
      }

      &_text_gray {
        color: $color-gray-5;
      }
    }

    &-count {
      font-size: $font-size-xxxl;

      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }
}
