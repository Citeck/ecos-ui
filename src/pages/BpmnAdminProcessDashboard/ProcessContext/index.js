import React, { useEffect, useState } from 'react';

import get from 'lodash/get';
import * as queryString from 'query-string';

import { decodeLink, getLastPathSegmentBeforeQuery, getSearchParams, replaceHistoryLink } from '../../../helpers/urls';
import { URL } from '../../../constants';
import { getValue } from '../utils';
import { PROCESS_TABS_TYPES } from '../../../constants/processAdmin';
import { getKeyProcessBPMN } from '../../../helpers/util';

export const ProcessContext = React.createContext();

export const ProcessContextProvider = props => {
  const urlParams = getSearchParams();
  const typeSchema = getLastPathSegmentBeforeQuery();
  const { recordRef, version } = urlParams;

  const [selectedVersion, setSelectedVersion] = useState(null);
  const [processId, setProcessId] = useState(recordRef);
  const [activityElement, setActivityElement] = useState();
  const [activeTabId, setActiveTabId] = useState(PROCESS_TABS_TYPES.INSTANCES);

  useEffect(
    () => {
      if (selectedVersion && typeSchema === URL.BPMN_ADMIN_PROCESS) {
        const newRecordRef = selectedVersion && (selectedVersion.processId || selectedVersion.id);
        if (getKeyProcessBPMN(recordRef) === getKeyProcessBPMN(newRecordRef)) {
          setProcessId(newRecordRef);
          replaceHistoryLink(
            undefined,
            `${URL.BPMN_ADMIN_PROCESS}?${decodeLink(
              queryString.stringify({
                ...urlParams,
                recordRef: newRecordRef,
                version: getValue(selectedVersion)
              })
            )}`,
            true
          );
        }
      }
    },
    [selectedVersion, typeSchema]
  );

  return (
    <ProcessContext.Provider
      value={{
        processId,
        processKey: get(selectedVersion, 'key'),
        selectedVersion,
        setSelectedVersion,

        activityElement,
        setActivityElement,

        activeTabId,
        setActiveTabId
      }}
    >
      {props.children}
    </ProcessContext.Provider>
  );
};
