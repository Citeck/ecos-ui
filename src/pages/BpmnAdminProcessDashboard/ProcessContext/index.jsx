import get from 'lodash/get';
import * as queryString from 'query-string';
import React, { useEffect, useState } from 'react';

import { URL } from '../../../constants';
import { PROCESS_TABS_TYPES } from '../../../constants/processAdmin';
import { decodeLink, getLastPathSegmentBeforeQuery, getSearchParams, replaceHistoryLink } from '../../../helpers/urls';
import { getKeyProcessBPMN } from '../../../helpers/util';
import { getValue } from '../utils';

export const ProcessContext = React.createContext();

export const ProcessContextProvider = props => {
  const urlParams = getSearchParams();
  const typeSchema = getLastPathSegmentBeforeQuery();
  const { recordRef } = urlParams;

  const [selectedVersion, setSelectedVersion] = useState(null);
  const [processId, setProcessId] = useState(recordRef);
  const [activityElement, setActivityElement] = useState();
  const [activeTabId, setActiveTabId] = useState(PROCESS_TABS_TYPES.INSTANCES);

  useEffect(() => {
    if (selectedVersion && typeSchema === URL.BPMN_ADMIN_PROCESS) {
      const newRecordRef = selectedVersion && (selectedVersion.processId || selectedVersion.id);

      setProcessId(newRecordRef);

      replaceHistoryLink(
        undefined,
        `${URL.BPMN_ADMIN_PROCESS}?${decodeLink(
          queryString.stringify({
            ...urlParams,
            recordRef: newRecordRef,
            version: getValue(selectedVersion)
          })
        )}`,
        true
      );
    }
  }, [selectedVersion, typeSchema]);

  return (
    <ProcessContext.Provider
      value={{
        processId,
        processKey: get(selectedVersion, 'key'),
        selectedVersion,
        setSelectedVersion,

        activityElement,
        setActivityElement,

        activeTabId,
        setActiveTabId
      }}
    >
      {props.children}
    </ProcessContext.Provider>
  );
};
