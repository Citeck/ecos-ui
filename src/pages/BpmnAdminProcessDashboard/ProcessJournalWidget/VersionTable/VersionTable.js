import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';

import get from 'lodash/get';
import isFunction from 'lodash/isFunction';
import head from 'lodash/head';

import { CommonTable } from '../../../../components/CommonTable';
import { Loader } from '../../../../components/common';
import { getTableColumns } from './constants';
import {
  getJournalTabInfo,
  setJournalTabInfoFilters,
  setJournalTabInfoPage,
  setJournalTabInfoSortBy
} from '../../../../actions/processAdmin';
import { selectProcessTabInfo } from '../../../../selectors/processAdmin';

const VersionTable = ({ isMobile, processId, dataInfo, getDataInfo, setPage, setSortBy, setFilters, tabId }) => {
  const [journalId, setJournalId] = useState(null);

  useEffect(
    () => {
      isFunction(getDataInfo) && getDataInfo(processId, tabId);
    },
    [processId, tabId]
  );

  useEffect(
    () => {
      if (!dataInfo || !dataInfo.data) {
        return;
      }

      const firstData = head(dataInfo.data);
      const [, journalId] = (get(firstData, 'documentJournalId') || '').split('@');

      if (journalId) {
        setJournalId(journalId);
      }
    },
    [dataInfo]
  );

  if (!dataInfo || !dataInfo.data) {
    return <Loader />;
  }

  const handleChangePage = ({ page, maxItems }) => {
    const newPage = {
      ...dataInfo.page,
      page,
      maxItems,
      skipCount: (page - 1) * maxItems
    };

    isFunction(setPage) && setPage(processId, newPage, tabId);
    isFunction(getDataInfo) && getDataInfo(processId, tabId);
  };

  const handleSort = e => {
    const sortBy = {
      attribute: e.column.attribute,
      ascending: !e.ascending
    };

    isFunction(setSortBy) && setSortBy(processId, sortBy, tabId);
    isFunction(getDataInfo) && getDataInfo(processId, tabId);
  };

  const handleFilter = filters => {
    const preparedFilters = filters.map(f => {
      if (f.att === 'document') {
        return {
          ...f,
          val: head(f.val)
        };
      }

      if (f.att === 'message') {
        return {
          ...f,
          t: 'like'
        };
      }

      return f;
    });

    isFunction(setFilters) && setFilters(processId, preparedFilters, tabId);
    isFunction(getDataInfo) && getDataInfo(processId, tabId);
  };

  const data = dataInfo.data.map(instance => ({
    ...instance,
    incidentCount: get(instance, 'incidents.length'),
    reload: getDataInfo
  }));

  return (
    <CommonTable
      totalCount={dataInfo.totalCount}
      isMobile={isMobile}
      isLoading={dataInfo.loading}
      data={data}
      columns={getTableColumns(tabId, { tabId, documentJournalId: journalId })}
      pagination={dataInfo.page}
      handleChangePage={handleChangePage}
      sortBy={dataInfo.sortBy}
      onSort={handleSort}
      filters={dataInfo.filters}
      onFilter={handleFilter}
      withoutSearch
      filterable
    />
  );
};

const mapStateToProps = (store, props) => {
  return {
    isMobile: store.view.isMobile,
    dataInfo: selectProcessTabInfo(store, props)
  };
};

const mapDispatchToProps = dispatch => ({
  getDataInfo: (processId, tabId) => dispatch(getJournalTabInfo({ tabId, processId })),
  setPage: (processId, page, tabId) => dispatch(setJournalTabInfoPage({ tabId, processId, page })),
  setSortBy: (processId, sortBy, tabId) => dispatch(setJournalTabInfoSortBy({ tabId, processId, sortBy })),
  setFilters: (processId, filters, tabId) => dispatch(setJournalTabInfoFilters({ tabId, processId, filters }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(VersionTable);
