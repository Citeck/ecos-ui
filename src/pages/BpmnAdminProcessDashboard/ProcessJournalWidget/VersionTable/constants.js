import React from 'react';

import { BpmnProcessStatusColumn } from '../../../../components/BpmnProcessStatusColumn';
import { createDocumentUrl } from '../../../../helpers/urls';
import { URL } from '../../../../constants';
import { ActivityComponent, MessageComponent } from '../components';
import { FailedActivityComponent } from '../components/FailedActivityComponent';
import { NoteComponent } from '../components/NoteComponent';
import { RetryFailedJob } from '../components/RetryFailedJob';
import { PROCESS_TABS_TYPES } from '../../../../constants/processAdmin';
import { getMLValue } from '../../../../helpers/util';

export const getTableColumns = (tabId, args) => {
  switch (tabId) {
    case PROCESS_TABS_TYPES.INSTANCES:
      return getInstancesTableColumns(args);
    case PROCESS_TABS_TYPES.INCIDENTS:
      return getIncidentsTableColumns(args);
    case PROCESS_TABS_TYPES.JOB_DEFINITIONS:
      return getProcessJobDefs(args);
    default:
      return [];
  }
};

export const getInstancesTableColumns = ({ documentJournalId }) => [
  {
    newFormatter: { type: 'link', config: { getUrl: row => `${URL.BPMN_ADMIN_INSTANCE}?recordRef=${row.id}` } },
    text: getMLValue({
      ru: 'ID',
      en: 'ID'
    }),
    attribute: 'id',
    dataField: 'dispId'
  },
  {
    newFormatter: { type: 'link', config: { getUrl: row => createDocumentUrl(row.documentId) } },
    newEditor: { type: 'journal', config: { journalId: documentJournalId, multiple: false } },
    searchable: true,
    searchableByText: true,
    disableSelect: !documentJournalId,
    text: getMLValue({
      ru: 'Документ',
      en: 'Document'
    }),
    attribute: 'document',
    dataField: 'document'
  },
  {
    newFormatter: { type: 'date', config: { format: 'LLL' } },
    sortable: true,
    text: getMLValue({
      ru: 'Время старта',
      en: 'Start time'
    }),
    attribute: 'startTime',
    dataField: 'startTime'
  },
  {
    classes: 'common-table-column__small',
    newFormatter: { type: 'component', config: { Component: BpmnProcessStatusColumn } },
    text: getMLValue({
      ru: 'Состояние',
      en: 'State'
    }),
    attribute: 'id',
    dataField: 'dispId'
  }
];

export const getIncidentsTableColumns = ({ tabId }) => [
  {
    newFormatter: { type: 'component', config: { Component: MessageComponent } },
    newEditor: { type: 'text', config: {} },
    text: getMLValue({
      ru: 'Сообщение',
      en: 'Message'
    }),
    searchable: true,
    searchableByText: true,
    attribute: 'message',
    dataField: 'message'
  },
  {
    newFormatter: { type: 'link', config: { getUrl: row => `${URL.BPMN_ADMIN_INSTANCE}?recordRef=${row.id}` } },
    text: getMLValue({
      ru: 'Экземпляр процесса',
      en: 'Process instance'
    }),
    sortable: true,
    attribute: 'id',
    dataField: 'dispId'
  },
  {
    newFormatter: { type: 'date', config: { format: 'LLL' } },
    sortable: true,
    text: getMLValue({
      ru: 'Дата создания',
      en: 'Created'
    }),
    attribute: 'created',
    dataField: 'created'
  },
  {
    newFormatter: { type: 'component', config: { Component: ActivityComponent } },
    sortable: true,
    text: getMLValue({
      ru: 'Элемент',
      en: 'Element'
    }),
    attribute: 'activityId',
    dataField: 'activityId'
  },
  {
    newFormatter: { type: 'component', config: { Component: FailedActivityComponent } },
    text: getMLValue({
      ru: 'Отказавший элемент',
      en: 'Failed element'
    }),
    attribute: 'failedActivityId',
    dataField: 'failedActivityId'
  },
  {
    sortable: true,
    text: getMLValue({
      ru: 'Тип',
      en: 'Type'
    }),
    attribute: 'incidentType',
    dataField: 'incidentType'
  },
  {
    classes: 'common-table-column__small',
    newFormatter: { type: 'component', config: { Component: NoteComponent } },
    text: getMLValue({
      ru: 'Примечание',
      en: 'Note'
    }),
    attribute: 'note',
    dataField: 'note'
  },
  {
    classes: 'common-table-column__small',
    newFormatter: { type: 'component', config: { Component: ({ row }) => <RetryFailedJob row={row} tabId={tabId} /> } },
    text: getMLValue({
      ru: 'Действия',
      en: 'Actions'
    }),
    attribute: 'id',
    dataField: 'id'
  }
];

export const getProcessJobDefs = () => [
  {
    text: getMLValue({
      ru: 'Состояние',
      en: 'State'
    }),
    attribute: 'state',
    dataField: 'state'
  },
  {
    newFormatter: { type: 'component', config: { Component: ActivityComponent } },
    text: getMLValue({
      ru: 'Элемент',
      en: 'Element'
    }),
    attribute: 'activityId',
    dataField: 'activityId'
  },
  {
    text: getMLValue({
      ru: 'Тип',
      en: 'Type'
    }),
    attribute: 'type',
    dataField: 'type'
  },
  {
    text: getMLValue({
      ru: 'Конфигурация',
      en: 'Configuration'
    }),
    attribute: 'configuration',
    dataField: 'configuration'
  }
];
