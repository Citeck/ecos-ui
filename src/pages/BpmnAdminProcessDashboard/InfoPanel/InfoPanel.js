import React from 'react';
import { connect } from 'react-redux';

import get from 'lodash/get';
import isFunction from 'lodash/isFunction';
import moment from 'moment';

import PanelTitle, { COLOR_GRAY } from '../../../components/common/PanelTitle/PanelTitle';
import { Loader } from '../../../components/common';
import Dashlet from '../../../components/Dashlet';
import BaseWidget from '../../../components/widgets/BaseWidget';
import { selectProcessMetaInfo, selectProcessVersions } from '../../../selectors/processAdmin';
import { getMetaInfo } from '../../../actions/processAdmin';
import { createDocumentUrl } from '../../../helpers/urls';
import { Btn } from '../../../components/common/btns';
import PageService from '../../../services/PageService';
import { t } from '../../../helpers/util';
import VersionSelect from '../VersionSelect';
import { Labels } from './constants';

import './style.scss';

class InfoPanel extends BaseWidget {
  componentDidMount() {
    const { processId, getMetaInfo } = this.props;

    isFunction(getMetaInfo) && getMetaInfo(processId);
  }

  componentDidUpdate(prevProps) {
    const { processId, getMetaInfo } = this.props;

    if (processId !== prevProps.processId) {
      isFunction(getMetaInfo) && getMetaInfo(processId);
    }
  }

  handleDefClick = () => {
    const { processId } = this.props;
    PageService.changeUrlLink(createDocumentUrl(processId), {
      openNewTab: true
    });
  };

  renderInfo = attribute => {
    const { metaInfo } = this.props;

    return metaInfo[attribute] || t(Labels.NOT_DEFINED);
  };

  render() {
    const { versions, metaInfo, processId } = this.props;

    const showLoader = !metaInfo || metaInfo.loading;
    const showVersionsLoading = get(versions, 'loading');

    return (
      <Dashlet
        title={t(Labels.INFO_TITLE)}
        className="info-panel"
        needGoTo={false}
        onToggleCollapse={this.handleToggleContent}
        isCollapsed={this.isCollapsed}
        setRef={this.setDashletRef}
      >
        {showLoader && <Loader />}
        {!showLoader && (
          <div className="info-panel__body">
            <div className="info-panel__main">
              <div className="info-panel__main-item">
                <PanelTitle narrow color={COLOR_GRAY}>
                  {t(Labels.VERSION)}
                </PanelTitle>
                <VersionSelect processId={processId} />
              </div>
              <div className="info-panel__main-item">
                <PanelTitle narrow color={COLOR_GRAY}>
                  {t(Labels.DEFINITION_REF)}
                </PanelTitle>
                <Btn className="ecos-btn_blue" onClick={this.handleDefClick} disabled={!metaInfo.definitionRef}>
                  {metaInfo.definitionRef || t(Labels.NOT_DEFINED)}
                  <i className="icon-small-right" />
                </Btn>
              </div>
            </div>

            <div className="info-panel__secondary">
              <span className="info-panel__main-item info-panel__main-item_full_width">{`${t(Labels.MODIFIER)}: ${metaInfo.modifier ||
                t(Labels.NOT_DEFINED)}`}</span>
              <span className="info-panel__main-item info-panel__main-item_full_width">{`${t(Labels.MODIFIED)}: ${
                metaInfo.modified ? moment(metaInfo.modified).format('LLL') : t(Labels.NOT_DEFINED)
              }`}</span>
            </div>

            <div className="info-panel__statistic">
              <div className="info-panel__statistic-item">
                <h5>{`${t(Labels.INSTANCES_COUNT)}:`}</h5>
                <div className="info-panel__statistic-count">{get(metaInfo, 'statistics.instancesCount')}</div>
              </div>
              <div className="info-panel__statistic-item">
                <h5>{`${t(Labels.INCIDENT_COUNT)}:`}</h5>
                <div className="info-panel__statistic-count">{get(metaInfo, 'statistics.incidentCount')}</div>
              </div>
              <div className="info-panel__statistic-item">
                <h5>{`${t(Labels.VERSIONS_COUNT)}:`}</h5>
                <div className="info-panel__statistic-count">
                  {showVersionsLoading ? <Loader type="points" /> : get(versions, 'data.length')}
                </div>
              </div>
              <div className="info-panel__statistic-item">
                <h5>{`${t(Labels.INSTANCES_COUNT)}:`}</h5>
                <span className="info-panel__statistic-item_text_gray">{t(Labels.BY_ALL_VERSIONS)}</span>
                <div className="info-panel__statistic-count">{metaInfo.allRunningInstancesCount}</div>
              </div>
            </div>
          </div>
        )}
      </Dashlet>
    );
  }
}

const mapStateToProps = (store, props) => {
  return {
    metaInfo: selectProcessMetaInfo(store, props),
    versions: selectProcessVersions(store, props)
  };
};

const mapDispatchToProps = dispatch => ({
  getMetaInfo: processId => dispatch(getMetaInfo({ processId }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(InfoPanel);
