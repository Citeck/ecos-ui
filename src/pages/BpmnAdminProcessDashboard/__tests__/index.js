import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { getIncidentsTableColumns, getInstancesTableColumns, getProcessJobDefs } from '../ProcessJournalWidget/VersionTable/constants';
import { ProcessContextProvider } from '../ProcessContext';
import InfoPanel from '../InfoPanel';
import configureStore from '../../../store';
import { PROCESS_TABS_TYPES } from '../../../constants/processAdmin';

const initialState = {};

const tableProps = {
  documentJournalId: 'document-journal-id',
};

const renderWithRedux = (Component) => {
  const store = configureStore(null, { bpmnAdmin: initialState });

  return (
    <Provider store={store}>
      <ProcessContextProvider>
        <Component />
      </ProcessContextProvider>
    </Provider>
  );
};

describe('', () => {
  describe('', () => {
    it('should render InfoPanel component', () => {
      const { container } = render(renderWithRedux(() => <InfoPanel processId="test-process-id" />));
      expect(container).toMatchSnapshot();
    });

    it('should render corrent columns for instances', () => {
      const columns = getInstancesTableColumns({ ...tableProps, tabId: PROCESS_TABS_TYPES.INSTANCES });
      expect(columns).toMatchSnapshot();
    });

    it('should render corrent columns for incidents', () => {
      const columns = getIncidentsTableColumns({ ...tableProps, tabId: PROCESS_TABS_TYPES.INCIDENTS });
      expect(columns).toMatchSnapshot();
    });

    it('should render corrent columns for job definitions', () => {
      const columns = getProcessJobDefs({ ...tableProps, tabId: PROCESS_TABS_TYPES.JOB_DEFINITIONS });
      expect(columns).toMatchSnapshot();
    });
  });
});
