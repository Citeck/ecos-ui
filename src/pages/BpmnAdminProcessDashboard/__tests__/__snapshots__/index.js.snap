// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`  should render InfoPanel component 1`] = `
<div>
  <div
    class="dashlet"
  >
    <div
      class="ecos-well ecos-panel dashlet info-panel"
    >
      <div
        class="ecos-panel__head dashlet__header-wrapper"
      >
        <div
          class="dashlet__header"
        >
          <span
            class="dashlet__caption"
          >
            bpmn-admin.info-title
          </span>
          <span
            class="ecos-badge ecos-badge_info ecos-badge_large"
          />
          <div
            class="dashlet__header-actions"
          />
        </div>
      </div>
      <div
        class="ecos-panel__body dashlet__body"
      >
        <div
          class="dashlet__body-content"
        >
          <div
            class="ecos-loader"
          >
            <div>
              <svg
                height="45"
                stroke="#7396cd"
                viewBox="0 0 38 38"
                width="45"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <g
                    stroke-width="2"
                    transform="translate(1 1)"
                  >
                    <circle
                      cx="18"
                      cy="18"
                      r="18"
                      stroke-opacity=".5"
                    />
                    <path
                      d="M36 18c0-9.94-8.06-18-18-18"
                    >
                      <animatetransform
                        attributeName="transform"
                        dur="1s"
                        from="0 18 18"
                        repeatCount="indefinite"
                        to="360 18 18"
                        type="rotate"
                      />
                    </path>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="dashlet__body-indent dashlet__body-indent_bottom"
        />
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`  should render corrent columns for incidents 1`] = `
[
  {
    "attribute": "message",
    "dataField": "message",
    "newEditor": {
      "config": {},
      "type": "text",
    },
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "searchable": true,
    "searchableByText": true,
    "text": "Message",
  },
  {
    "attribute": "processInstanceRef",
    "dataField": "processInstanceRef",
    "newFormatter": {
      "config": {
        "getUrl": [Function],
      },
      "type": "link",
    },
    "sortable": true,
    "text": "Process instance",
  },
  {
    "attribute": "created",
    "dataField": "created",
    "newFormatter": {
      "config": {
        "format": "LLL",
      },
      "type": "date",
    },
    "sortable": true,
    "text": "Created",
  },
  {
    "attribute": "activityId",
    "dataField": "activityId",
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "sortable": true,
    "text": "Element",
  },
  {
    "attribute": "failedActivityId",
    "dataField": "failedActivityId",
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Failed element",
  },
  {
    "attribute": "incidentType",
    "dataField": "incidentType",
    "sortable": true,
    "text": "Type",
  },
  {
    "attribute": "note",
    "classes": "common-table-column__small",
    "dataField": "note",
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Note",
  },
  {
    "attribute": "id",
    "classes": "common-table-column__small",
    "dataField": "id",
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Actions",
  },
]
`;

exports[`  should render corrent columns for instances 1`] = `
[
  {
    "attribute": "id",
    "dataField": "dispId",
    "newFormatter": {
      "config": {
        "getUrl": [Function],
      },
      "type": "link",
    },
    "text": "ID",
  },
  {
    "attribute": "document",
    "dataField": "document",
    "disableSelect": false,
    "newEditor": {
      "config": {
        "journalId": "document-journal-id",
        "multiple": false,
      },
      "type": "journal",
    },
    "newFormatter": {
      "config": {
        "getUrl": [Function],
      },
      "type": "link",
    },
    "searchable": true,
    "searchableByText": true,
    "text": "Document",
  },
  {
    "attribute": "startTime",
    "dataField": "startTime",
    "newFormatter": {
      "config": {
        "format": "LLL",
      },
      "type": "date",
    },
    "sortable": true,
    "text": "Start time",
  },
  {
    "attribute": "id",
    "classes": "common-table-column__small",
    "dataField": "dispId",
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "State",
  },
]
`;

exports[`  should render corrent columns for job definitions 1`] = `
[
  {
    "attribute": "state",
    "dataField": "state",
    "text": "State",
  },
  {
    "attribute": "activityId",
    "dataField": "activityId",
    "newFormatter": {
      "config": {
        "Component": [Function],
      },
      "type": "component",
    },
    "text": "Element",
  },
  {
    "attribute": "type",
    "dataField": "type",
    "text": "Type",
  },
  {
    "attribute": "configuration",
    "dataField": "configuration",
    "text": "Configuration",
  },
]
`;
